src/api/forms/service.js

```diff
@@ -189,6 +189,70 @@ export async function updateDraftFormDefinition(formId, definition, author) {
   }
 }
 
+/**
+ * @param {string} formId - ID of the form
+ * @param {FormMetadataInput} formUpdate - full JSON form definition
+ * @param {FormMetadataAuthor} author - the author details
+ * @returns {Promise<string>}
+ */
+export async function updateFormMetadata(formId, formUpdate, author) {
+  logger.info(`Updating form metadata (draft) for form ID ${formId}`)
+
+  try {
+    // Get the form metadata from the db
+    const form = await getForm(formId)
+
+    if (form.live && formUpdate.title) {
+      throw Boom.badRequest(
+        `Field ${formUpdate.title} cannot be updated once the form has gone live`
+      )
+    }
+
+    const updatedSlug = formUpdate.title && slugify(formUpdate.title)
```

Comment:
We only want to update the slug if  `formUpdate.title` is provided, this will set the updated slug to `undefined` if title is missing in the update request.

Reply:
I thought about this but didn't fix it yet. I will fix it during my testing 👍